### DISCUSSION TITLE
How to Replicate Figures 4 and 5 from the Paper "Calculating Anisotropic Piezoelectric Properties from Texture Data using the MTEX Open Source Package"?

### DISCUSSION THREAD

**Discussion by obaica** [mtex-toolbox/mtex#2208] [Ask Anything]
Hello MTEX team,

I am a new user of MTEX and I am trying to replicate Figures 4 and 5 from the paper titled "Calculating anisotropic piezoelectric properties from texture data using the MTEX open source package."

I attempted to generate these figures using the following code:

```
CS = crystalSymmetry('32', [4.916 4.916 5.4054], 'X||a*', 'Z||c', 'mineral', 'Quartz');

fname = fullfile(mtexDataPath,'tensor', 'Single_RH_quartz_poly.P');

P = tensor.load(fname,CS,'propertyname','piecoelectricity','unit','C/N','DoubleConvention');

% Generate a 3D plot
f = figure;
plot(P, '3d');
rotate3d;
```

However, I am unable to obtain the correct results as shown in the paper. Could you please provide guidance or the specific code needed to reproduce Figures 4 and 5 from the mentioned paper?

Thank you for your assistance!

Best regards,

**Comment by lfgmorales** [comment]
Hi

Can you share the figure you get?

Cheers

Luiz
Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 2 Sep 2024, at 17:17, Gang Tang ***@***.***> wrote:

﻿

Hello MTEX team,

I am a new user of MTEX and I am trying to replicate Figures 4 and 5 from the paper titled "Calculating anisotropic piezoelectric properties from texture data using the MTEX open source package."

I attempted to generate these figures using the following code:

`CS = crystalSymmetry('32', [4.916 4.916 5.4054], 'X||a*', 'Z||c', 'mineral', 'Quartz');

fname = fullfile(mtexDataPath,'tensor', 'Single_RH_quartz_poly.P');

P = tensor.load(fname,CS,'propertyname','piecoelectricity','unit','C/N','DoubleConvention');

% Generate a 3D plot
f = figure;
plot(P, '3d');
rotate3d;
`
However, I am unable to obtain the correct results as shown in the paper. Could you please provide guidance or the specific code needed to reproduce Figures 4 and 5 from the mentioned paper?

Thank you for your assistance!

Best regards,

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2208>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN2VXWZ3BEFOXX2GGRDZUR6P3AVCNFSM6AAAAABNQQ5GW2VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGEZDQNZUGU>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2208", "url": "https://github.com/mtex-toolbox/mtex/discussions/2208", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by lfgmorales** [comment]
You need to plot the property related to P, for example P.directionalMagnitude

Look here

<https://mtex-toolbox.github.io/PiezoElectricity.html>
The Piezoelectricity Tensor | MTEX<https://mtex-toolbox.github.io/PiezoElectricity.html>
mtex-toolbox.github.io<https://mtex-toolbox.github.io/PiezoElectricity.html>
        [favicon.png] <https://mtex-toolbox.github.io/PiezoElectricity.html>



Cheers

L


Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 2 Sep 2024, at 17:27, Grafulha Morales Luiz ***@***.***> wrote:

﻿ Hi

Can you share the figure you get?

Cheers

Luiz
Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 2 Sep 2024, at 17:17, Gang Tang ***@***.***> wrote:

﻿

Hello MTEX team,

I am a new user of MTEX and I am trying to replicate Figures 4 and 5 from the paper titled "Calculating anisotropic piezoelectric properties from texture data using the MTEX open source package."

I attempted to generate these figures using the following code:

`CS = crystalSymmetry('32', [4.916 4.916 5.4054], 'X||a*', 'Z||c', 'mineral', 'Quartz');

fname = fullfile(mtexDataPath,'tensor', 'Single_RH_quartz_poly.P');

P = tensor.load(fname,CS,'propertyname','piecoelectricity','unit','C/N','DoubleConvention');

% Generate a 3D plot
f = figure;
plot(P, '3d');
rotate3d;
`
However, I am unable to obtain the correct results as shown in the paper. Could you please provide guidance or the specific code needed to reproduce Figures 4 and 5 from the mentioned paper?

Thank you for your assistance!

Best regards,

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2208>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN2VXWZ3BEFOXX2GGRDZUR6P3AVCNFSM6AAAAABNQQ5GW2VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGEZDQNZUGU>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2208", "url": "https://github.com/mtex-toolbox/mtex/discussions/2208", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by obaica** [comment]
Thank you for your reply.

I have read the page you provided and tried using the `surf(P.directionalMagnitude)` method. However, the resulting plot is clearly different from the one in the literature I mentioned. Moreover, the horizontal axis of the plot generated by the `surf(P.directionalMagnitude)` method ranges from -1 to 1 for any material, rather than reflecting the actual range of dij values."

![P.directionalMagnitude for a-quartz ](https://github.com/user-attachments/assets/45278d86-6f9f-4b8f-8209-8570350ec815)
![3D plot of the piezoelectric tensor d for right-handed a-quartz](https://github.com/user-attachments/assets/a86c8eb2-9ef6-4948-b8d3-7b157c58f002)

**Comment by obaica** [comment]
I would like to know how to use `MTEX` to obtain the plot shown below.

![image](https://github.com/user-attachments/assets/ca04e243-eeee-461f-8769-dd01980cf57e)
